packages/ra-core/src/controller/button/useDeleteWithConfirmController.tsx

```diff
@@ -72,27 +73,38 @@ const useDeleteWithConfirmController = ({
     redirect: redirectTo,
     basePath,
     onClick,
+    onSuccess,
+    onFailure,
 }: UseDeleteWithConfirmControllerParams): UseDeleteWithConfirmControllerReturn => {
     const [open, setOpen] = useState(false);
     const notify = useNotify();
     const redirect = useRedirect();
     const refresh = useRefresh();
-    const [deleteOne, { loading }] = useDelete(resource, null, null, {
-        action: CRUD_DELETE,
-        onSuccess: () => {
-            setOpen(false);
+    if (onSuccess === undefined) {
+        onSuccess = () => {
```

Comment:
you should not change a parameter passed to the hook. Prefer inlining the test for the `onSuccess` inside the `useDelete` call.